portfolio/src/main/webapp/script.js

```diff
@@ -80,13 +80,68 @@ function openCity(evt, sectionName) {
 /**
  * Fetch function from servlet
  */
-const START_INDEX = 4;
-const END_INDEX = 17;
 async function getWelcomeData() {
-  const response = await fetch('/data');
-  let quote = await response.text();
-  // Slice quote so that formatting remains when on navigates to url/data
-  document.getElementById('data-container').innerText = quote.slice(START_INDEX, END_INDEX);
+    // Get comments from the server
+    const response = await fetch('/data');
+    let comments = await response.json();
+
+    // Display the comments
+    const commentsEl = document.getElementById('comment_list');
+    comments.forEach((comment) => {
+        commentsEl.appendChild(createCommentElement(comment));
+    });
+}
+
+/**
+ * Creates one comment element
+ */
+function createCommentElement(comment) {
+    const commentElement = document.createElement('li');
+    commentElement.className = 'comment';
+
+    const titleElement = document.createElement('span');
+    titleElement.className = "comment_title";
+    titleElement.innerText = comment.name;
+
+    const deleteButtonElement = document.createElement('button');
+    deleteButtonElement.className = "comment_button delete"
+    deleteButtonElement.innerHTML = '<i class="fa fa-times"></i>';
```

Comment:
Using `innerHTML` is inherently unsafe from a security point of view, because it's possible for text crafted by a malicious user of your site to end up being parsed as HTML, which would let them take control over the site.  It's called a cross-site scripting (XSS) attack, and the best way to prevent it is to do what you're doing elsewhere in this function and construct your DOM with `createElement` instead.  That said, I notice you've added jQuery to your site, and jQuery has some great APIs for quickly constructing DOM hierarchies that are almost as convenient as using `innerHTML` without being evil.  🙂

Reply:
great tip!